src/Bitcoin/InputSelector.cpp

```diff
@@ -92,43 +103,43 @@ UTXOs UnspentSelector::select(const UTXOs& utxos, int64_t targetValue, int64_t b
     //    (1) bigger than what we need
     //    (2) closer to 2x the amount,
     //    (3) and does not produce dust change.
-    for (int64_t numInputs = 1; numInputs <= n; ++numInputs) {
+    for (size_t numInputs = 1; numInputs <= n; ++numInputs) {
         const auto fee = feeCalculator.calculate(numInputs, numOutputs, byteFee);
         const auto targetWithFeeAndDust = targetValue + fee + dustThreshold;
         if (maxWithXInputs[numInputs] < targetWithFeeAndDust) {
             // no way to satisfy with only numInputs inputs, skip
             continue;
         }
-        auto slices = slice(sortedUtxos, static_cast<size_t>(numInputs));
+        auto slices = slice(sortedIndices, static_cast<size_t>(numInputs));
+
         slices.erase(
             std::remove_if(slices.begin(), slices.end(),
```

Comment:
let's add a `filter` help method? I wanted to use it too

Reply:
Filter what?  Below a limit, or dust? `filterDustInput` and `filterThreshold` is available, I'll make them public.
https://github.com/trustwallet/wallet-core/pull/1650/files#diff-f308f664e0f47afd7ee448c0815cac953b53197567a0b86dcc9fe4de23ab81a8R41